src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -812,28 +812,38 @@ static void close_transport_locked(grpc_chttp2_transport* t,
           t->settings_ack_watchdog,
           grpc_event_engine::experimental::EventEngine::TaskHandle::kInvalid));
     }
-    if (t->delayed_ping_timer_handle.has_value()) {
-      if (t->event_engine->Cancel(*t->delayed_ping_timer_handle)) {
-        t->delayed_ping_timer_handle.reset();
+    if (t->delayed_ping_timer_handle !=
```

Comment:
Drive-by comment: could be more readable with something like:

```
// Added somewhere file-scoped, up above
using grpc_event_engine::experimental::EventEngine::TaskHandle;
...
if (t->delayed_ping_timer_handle != TaskHandle::kInvalid &&
    t->event_engine->Cancel(t->delayed_ping_timer_handle)) {
  t->delayed_ping_timer_handle = TaskHandle::kInvalid;
}
```

Reply:
Done. @drfloob adding you as reviewer since I take this as volunteering :)